components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/MarketplaceAssistantApiPublisherNotifier.java

```diff
@@ -106,51 +106,77 @@ private void process (Event event) throws NotifierException {
                 default:
                     break;
             }
+        } else if (APIConstants.EventType.API_UPDATE.name().equals(event.getType())) {
+            String currentStatus = apiEvent.getApiStatus().toUpperCase();
+            switch (currentStatus) {
+                case APIConstants.PROTOTYPED:
+                case APIConstants.PUBLISHED:
+                    postRequest(apiEvent);
+                    break;
+                default:
+                    break;
+            }
         }
     }
 
     private void postRequest(APIEvent apiEvent) throws NotifierException {
-        apiMgtDAO = ApiMgtDAO.getInstance();
         String apiId = apiEvent.getUuid();
 
         try {
+            apiMgtDAO = ApiMgtDAO.getInstance();
             APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(CarbonContext.
                     getThreadLocalCarbonContext().getUsername());
             API api = apiProvider.getAPIbyUUID(apiId, apiMgtDAO.getOrganizationByAPIUUID(apiId));
-            String api_type = api.getType();
-            JSONObject payload = new JSONObject();
 
-            switch (api_type) {
-                case APIConstants.API_TYPE_GRAPHQL:
-                    payload.put(APIConstants.API_SPEC_TYPE_GRAPHQL, api.getGraphQLSchema());
-                    payload.put(APIConstants.API_SPEC_TYPE, APIConstants.API_TYPE_GRAPHQL);
-                    break;
-                case APIConstants.API_TYPE_ASYNC:
-                case APIConstants.API_TYPE_WS:
-                case APIConstants.API_TYPE_WEBSUB:
-                case APIConstants.API_TYPE_SSE:
-                case APIConstants.API_TYPE_WEBHOOK:
-                    payload.put(APIConstants.API_SPEC_TYPE_ASYNC, api.getAsyncApiDefinition());
-                    payload.put(APIConstants.API_SPEC_TYPE, APIConstants.API_TYPE_ASYNC);
-                    break;
-                case APIConstants.API_TYPE_HTTP:
-                case APIConstants.API_TYPE_SOAP:
-                case APIConstants.API_TYPE_SOAPTOREST:
-                    payload.put(APIConstants.API_SPEC_TYPE_REST, api.getSwaggerDefinition());
-                    payload.put(APIConstants.API_SPEC_TYPE, APIConstants.API_TYPE_REST);
-                    break;
-                default:
-                    break;
-            }
+            Thread thread = new Thread(new Runnable() {
+                @Override
```

Comment:
Shall we create a runnable class and pass it here? And lets name the thread as well